platform/fio/transaction.go

```diff
@@ -30,28 +30,47 @@ func (p *Platform) GetTxsByAddress(address string) (page blockatlas.TxPage, err
 
 func (p *Platform) Normalize(action *Action, account string) (blockatlas.Tx, error) {
 	var (
-		to, from string
-		amount   blockatlas.Amount
+		to, from    string
+		amount, fee blockatlas.Amount
+		memo        string
 	)
 	const dateFormat string = "2006-01-02T15:04:05"
 
 	// Action type "transfer" handled (trnsfiopubky not)
 	if action.ActionTrace.Act.Account == "fio.token" &&
-		action.ActionTrace.Act.Name == "transfer" {
+		(action.ActionTrace.Act.Name == "transfer" || action.ActionTrace.Act.Name == "trnsfiopubky") {
 		// convert to action-specific data
-		var actionData ActionData
 		dataJSON, err := json.Marshal(action.ActionTrace.Act.Data)
 		if err != nil {
 			return blockatlas.Tx{}, errors.E("Unparseable Data field")
 		}
-		if json.Unmarshal(dataJSON, &actionData) != nil {
-			return blockatlas.Tx{}, errors.E("Unparseable Data field")
-		}
-		from = actionData.From
-		to = actionData.To
-		amountNum, err := strconv.ParseFloat(strings.Split(actionData.Quantity, " ")[0], 64)
-		if err == nil {
-			amount = blockatlas.Amount(strconv.Itoa(int(amountNum * 1000000000)))
+		switch action.ActionTrace.Act.Name {
+		case "transfer":
+			var actionData ActionDataTransfer
+			if json.Unmarshal(dataJSON, &actionData) != nil {
+				return blockatlas.Tx{}, errors.E("Unparseable Data field")
+			}
+			from = actionData.From
+			to = actionData.To
+			amountNum, err := strconv.ParseFloat(strings.Split(actionData.Quantity, " ")[0], 64)
```

Comment:
can we unmarshal as `Amount` directly?

Reply:
No, as it is a formatted string, doesn't make much sense
"200.000000000 FIO"